modelangelo/protocols/protocol_model_angelo.py

```diff
@@ -83,10 +80,53 @@ def _defineParams(self, form):
                        label="Use GPU for execution",
                        help="This protocol has both CPU and GPU implementation."
                             "Select the one you want to use.")
+                            
         form.addHidden(GPU_LIST, params.StringParam, default='0',
                        expertLevel=LEVEL_ADVANCED,
                        label="Choose GPU ID (single one)",
                        help="GPU device to be used")
+                       
+        form.addParam('configFile', params.FileParam,
```

Comment:
Do you see an intermediate case here?

Instead of providing a config file (keep this option), but providing some of the parameters in it and then, inside the protocol generate the config file based on those params? e.g:

form.addParam ("croplength",...


....

def writeConfig()
   json.write( ....) # using self.croplength.get()


¿?

Reply:
Hi @pconesa,

     I do not have a strong opinion here. So far I have only use the "croplength" variable so what you suggest would solve my problem. Anyway, I see this option as the  "ExtraParameters" option that we have in other protocols.
Just tell what way do you want to go I will rewrite the pull request.

Reply:
Hi @pconesa,

      What should I do with this pullrequest?

Reply:
Sorry, I'm too busy to take care of so many PR.

I'd say we can go ahead with this one.